File: Problems/Parking Lot/Java/payments/CarPaymentStrategy.java
package payments;

import java.time.Duration;

import ticket.Ticket;

public class CarPaymentStrategy implements Payments {
    private static final double CAR_RATE_PER_HOUR = 2.0; // Example rate per hour for car parking

    @Override
    public double pay(Ticket ticket) {
        Duration duration = Duration.between(ticket.getEntryTime(), ticket.getExitTime());

        double parkedHours = Math.ceil(duration.toSeconds() / 3600.0);

        return parkedHours * CAR_RATE_PER_HOUR;
    }
}

File: Problems/Parking Lot/Golang/payments/car_payment_strategy.go
package payments

import (
	"math"
	"parking-lot/ticket"
)

type CarPaymentStrategy struct{}

const (
	CAR_RATE_PER_HOUR = 2.0
)

func (cps *CarPaymentStrategy) pay(ticket ticket.Ticket) float64 {
	duration := ticket.GetExitTime().Sub(ticket.GetEntryTime())

	parkedHours := math.Ceil(duration.Seconds() / 3600.0)

	return parkedHours * CAR_RATE_PER_HOUR
}
